Javascript/100-Problems/script.js

```diff
@@ -652,3 +648,214 @@ const countStudentsScoredBelowMarkInAllSub = () => {
+      sumOfMarks = studentWithHighestTotal();
+      break;
+
+    case "studentLowestPercentage":
+      sumOfMarks = studentWithLowestTotal();
+      break;
+
+    case "subjectWithHighestPercentage":
+      sumOfMarks = highestTotalMark();
+      break;
+
+    case "subjectWithLowestPercentage":
+      sumOfMarks = lowestTotalMark();
+      break;
+  }
+
+  sumOfMarks.map((mark) => {
+    name = mark.name;
+    totalObtained = mark.totalObtained;
+  });
+
+  const percentageObtained = percentage(totalObtained, totalMark);
+
+  return { name, percentageObtained };
+};
+
+//Q35: Write a function to find and print the student(s) with the highest percentage of marks.
+
+const studentWithHighestPercentage = () => {
+  const operation = "studentHighestPercentage";
+  const totalMark = 250;
+
+  return studentWithPercentage(operation, totalMark);
+};
+
+console.log(studentWithHighestPercentage());
+
+//Q36: Write a function to find and print the student(s) with the lowest percentage of marks.
+
+const studentWithLowestPercentage = () => {
+  const operation = "studentLowestPercentage";
+  const totalMark = 250;
+
+  return studentWithPercentage(operation, totalMark);
+};
+
+console.log(studentWithLowestPercentage());
+
+//Q37: Write a function to find and print the subject(s) with the highest percentage of marks.
+
+const subjectWithHighestPercentage = () => {
+  const operation = "subjectWithHighestPercentage";
+  const totalMark = 200;
+
+  return studentWithPercentage(operation, totalMark);
+};
+
+console.log(subjectWithHighestPercentage());
+
+//Q38: Write a function to find and print the subject(s) with the lowest percentage of marks.
+
+const subjectWithLowestPercentage = () => {
+  const operation = "subjectWithLowestPercentage";
+  const totalMark = 200;
+
+  return studentWithPercentage(operation, totalMark);
+};
+
+console.log(subjectWithLowestPercentage());
+
+//function to find and print the student(s) with the highest & lowest percentage of marks in a specific subject.
+
+const studentWithPercentageInSubject = (total, operation, subjectName) => {
+  const students = classObj.students;
+  const totalMark = 50;
+  let percentages = 0;
+  let studentName = "";
+  let subjectMark = 0;
+
+  const markArray = students.map((student) => {
+    const subjects = student.marks;
+    studentName = student.name;
+
+    subjects.find((subject) => {
+      let mark = subject.mark;
+
+      if (subject.subject === subjectName) {
+        subjectMark = mark;
+
+        switch (operation) {
+          case "highest":
+            if (mark >= total) {
+              total = mark;
+            }
+            break;
+
+          case "lowest":
+            if (mark <= total) {
+              total = mark;
+            }
+            break;
+        }
+      }
+    });
+
+    percentages = percentage(subjectMark, totalMark);
+
+    return { studentName, subjectName, percentages };
+  });
+
+  total = percentage(total, totalMark);
```

Comment:
If we assign percentage to this variable named total, then its meaning changed right?. So I think we can give this directly.